### ISSUE TITLE
Sync Admin actions from/to trusted instances

### ISSUE THREAD

**Issue by jgrim** [sublinks/sublinks-api#137]
Instance 1 (Trusted with Instance 2) Admin Bans UserY@Instance2.example

Instance 2 (Trusted with Instance 1) Local UserY gets banned.

Instance 2 can select on how many trusted instances has to ban the user to ban the user localy

(Admins cant get banned through this)

**Comment by lazyguru** [comment]
I don't think it should be possible to cause a user to be banned based on remote instance actions. Maybe those remote instances have different policies than the local instance does? The remote instances are within their rights to kick the user from any communities _hosted on their instance_, but it should not be possible to cause the user to be banned from their own instance

**Comment by jgrim** [comment]
This would be a feature to allow other instances to moderate your instance for you. It wouldn't be on for everyone by default.

I would trust the top 3 instances for example to apply my rules.

**Comment by kersploosh** [comment]
As an alternative to automatically duplicating bans, allow a notification option. A local admin could configure each trusted instance to (a) duplicate the trusted instance's bans locally, or (b) notify the local admin of bans occurring on the trusted instance so they can be reviewed before taking action.

**Comment by jgrim** [comment]
I like that idea @kersploosh, thanks a lot!

**Comment by lazyguru** [comment]
This is a good idea, but maybe it should be done via federation? We could add a new apub message type to support this.

**Comment by gaviscapes** [comment]
I cringe a bit at the idea of having it fully automated, I agree with @kersploosh that it should be at least reviewed before hand or provide some sort of notification when it does the action. Attempts at automating blocklists for example across mastodon instances has caused serious issues in the fediverse in the past.

**Comment by lazyguru** [comment]
I agree, however having the data be shared makes sense. Each instance admin could go through the list to approve. I would say the frontend (and API) should *not* allow for a "bulk approve" (though I'd be ok with a "bulk ignore/reject"). Any kind of bulk approve action would be pointless since people would just use it and defeat the point of a review.
